rs/registry/canister/src/mutations/node_management/do_remove_node_directly.rs

```diff
@@ -184,4 +231,160 @@ mod tests {
+            ),
+            insert(
+                make_node_operator_record_key(operator2_id),
+                operator_record_2.encode_to_vec(),
+            ),
+        ]);
+        // Add node owned by operator1 to registry
+        let (mutate_request, node_ids_and_dkg_pks) =
+            prepare_registry_with_nodes_and_node_operator_id(
+                1, /* mutation id */
+                1, /* node count */
+                operator1_id,
+            );
+        registry.maybe_apply_mutation_internal(mutate_request.mutations);
+        let node_id = node_ids_and_dkg_pks
+            .keys()
+            .next()
+            .expect("should contain at least one node ID")
+            .to_owned();
+
+        let payload = RemoveNodeDirectlyPayload { node_id };
+
+        registry.do_remove_node(payload, operator2_id);
+    }
+
+    #[test]
+    fn should_succeed_remove_node_compare_dc_and_node_provider() {
+        let mut registry = invariant_compliant_registry(0);
+        // Add node operator1 and operator2 records, both under the same provider
+        let operator1_id = PrincipalId::new_user_test_id(2000);
+        let operator2_id = PrincipalId::new_user_test_id(2001);
+        let operator_record_1 = NodeOperatorRecord {
+            node_operator_principal_id: operator1_id.to_vec(),
+            node_provider_principal_id: PrincipalId::new_user_test_id(3000).to_vec(),
+            dc_id: "dc1".to_string(),
+            node_allowance: 1,
+            ..Default::default()
+        };
+        let operator_record_2 = NodeOperatorRecord {
+            node_operator_principal_id: operator2_id.to_vec(),
+            node_provider_principal_id: PrincipalId::new_user_test_id(3000).to_vec(),
+            dc_id: "dc1".to_string(),
+            node_allowance: 1,
+            ..Default::default()
+        };
+        registry.maybe_apply_mutation_internal(vec![
+            insert(
+                make_node_operator_record_key(operator1_id),
+                operator_record_1.encode_to_vec(),
+            ),
+            insert(
+                make_node_operator_record_key(operator2_id),
+                operator_record_2.encode_to_vec(),
+            ),
+        ]);
+        // Add node owned by operator1 to registry
+        let (mutate_request, node_ids_and_dkg_pks) =
+            prepare_registry_with_nodes_and_node_operator_id(
+                1, /* mutation id */
+                1, /* node count */
+                operator1_id,
+            );
+        registry.maybe_apply_mutation_internal(mutate_request.mutations);
+        let node_id = node_ids_and_dkg_pks
+            .keys()
+            .next()
+            .expect("should contain at least one node ID")
+            .to_owned();
+
+        let payload = RemoveNodeDirectlyPayload { node_id };
+
+        // Should succeed because both operator1 and operator2 are under the same provider
+        registry.do_remove_node(payload, operator2_id);
```

Comment:
We would have caught the bug if we had more asserts in this test. This test merely asserts that do_remove_node does not panic, which is a very weak assertion. What we should have done is also verify what data was ACTUALLY COMMITTED by this call.